### ISSUE TITLE
How to generate images with segmentation maps

### ISSUE THREAD

**Issue by FriedRonaldo** [vglsd/OpenSESAME#4]
Hi, thanks for sharing the interesting work and its code.

I have a question about the way to generate images with segmentation maps.

I trained a model with "train_cityscapes.sh", however, there is no guide to generate an image with a map -- only removal and addition are provided.

After seeing the generated images by "test.py", I decided to mask the whole image and provide it to the generator with the segmentation map. But, it seems not to work. The mask is generated by "mask = self.get_mask(image.size(), torch.zeros_like(mask)" -- the latter line replaces 'mask' to the zeros.

Because I do not use "--segmentation_map" option, I thought it might work. (If I called segmentation_map, then the segmentation map is masked, too. And it should make the input zeros!)

The below image is an example.

![image](https://user-images.githubusercontent.com/23406491/104431447-3d4b5000-55cb-11eb-8e98-05283ab4274a.png)

And this is the image generated with the same segmentation map but not a whole mask.

![image](https://user-images.githubusercontent.com/23406491/104431571-623fc300-55cb-11eb-8a6f-dabc27aa1e49.png)

I will wait for your reply. Thanks!

---

EDIT
Now, I try training the model by providing the mask as "zeros". Then, the input is the segmentation map only.
By doing so, the model provides realistic images with the segmentation maps. Is it the right direction to train a model to enforce to generate images only with segmentation maps?

**Comment by entavelis** [comment]
Hello,

If I understand correctly you are trying to do what we refer as "Layout to Image Generation". The "train_cityscapes.sh" script is training the model for the task of image manipulation, that is, to generate _edits_ on an existing image rather than generating the whole thing. 

The way we do layout to image generation in the paper is by coupling our SESAME discriminator to the [SPADE](https://github.com/NVlabs/SPADE) Generator. So, I would suggest you do the same. We also provide pretrained models for this approach if you want to evaluate!

Best,
Evan

**Comment by FriedRonaldo** [comment]
Thanks for the reply!

To summary your suggestion, if I want to conduct the layout to image generation (input image is the semantic segmentation image and the output is RGB image), then, I should train a model with the SESAME discriminator and SPADE Generator.
And are the loss functions the same as SESAME?

Did I understand correctly?

Again, thanks!

**Comment by entavelis** [comment]
Yes, indeed they are! You only need to change the function NLayerDiscriminator to our SesameNLayerDiscriminator in the discriminator.py of the SPADE repo. 

Cheers!

**Comment by FriedRonaldo** [comment]
Thanks for the kind guidance!
